gradoop-examples/src/main/java/org/gradoop/benchmark/sampling/SamplingBenchmark.java

```diff
@@ -0,0 +1,421 @@
+import org.gradoop.flink.model.impl.operators.sampling.RandomEdgeSampling;
+import org.gradoop.flink.model.impl.operators.sampling.PageRankSampling;
+import org.gradoop.flink.model.impl.operators.sampling.SamplingAlgorithm;
+import org.gradoop.flink.util.GradoopFlinkConfig;
+import org.gradoop.utils.statistics.StatisticsRunner;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A dedicated program to evaluate sampling algorithms on the basis of ldbc generated social network
+ * graphs
+ */
+public class SamplingBenchmark extends AbstractRunner implements ProgramDescription {
+
+  /**
+   * Required option to declare the path to the directory containing csv files to be processed
+   */
+  private static final String OPTION_INPUT_PATH = "i";
+
+  /**
+   * Option to declare the path to output directory
+   */
+  private static final String OPTION_OUTPUT_PATH = "o";
+
+  /**
+   * Option to define to-be-evaluated sampling algorithm
+   */
+  private static final String OPTION_SELECTED_ALGORITHM = "a";
+
+  /**
+   * Option to declare list of parameters that are passed on to a constructor
+   */
+  private static final String OPTION_CONSTRUCTOR_PARAMS = "p";
+
+  /**
+   * Option to to set whether graph statistics are to be computed
+   */
+  private static final String OPTION_COMPUTE_STATISTICS = "s";
+
+  /**
+   * Used input path
+   */
+  private static String INPUT_PATH;
+
+  /**
+   * Used output path
+   */
+  private static String OUTPUT_PATH;
+
+  /**
+   * Output path default
+   */
+  private static final String OUTPUT_PATH_DEFAULT = "./sampling_benchmark/";
+
+  /**
+   * Output path suffix defining where resulting graph sample is written to
+   */
+  private static final String OUTPUT_PATH_GRAPH_SAMPLE_SUFFIX = "graph_sample/";
+
+  /**
+   * Output path suffix defining where resulting benchmark file is written to
+   */
+  private static final String OUTPUT_PATH_BENCHMARK_SUFFIX = "benchmark";
+
+  /**
+   * Output path suffix defining where input graph statistics are written to
+   */
+  private static final String GRAPH_STATISTICS_ORIGIN = "graph_statistics/origin/";
+
+  /**
+   * Output path suffix defining where sample graph statistics are written to
+   */
+  private static final String GRAPH_STATISTICS_SAMPLE = "graph_statistics/sample/";
+
+  /**
+   * Integer defining the sampling algorithm that is to be evaluated.
+   */
+  private static int SELECTED_ALGORITHM;
+
+  /**
+   * List of parameters that are used to instantiate the selected sampling algorithm.
+   */
+  private static String[] CONSTRUCTOR_PARAMS;
+
+  /**
+   * Boolean determining if graph statistics for input and sample graph shall be computed
+   */
+  private static boolean COMPUTE_STATISTICS;
+
+  /**
+   * Enum of available sampling algorithms. Enum is constructed with meta information about a given
+   * sampling algorithm like its name.
+   */
+  private enum Algorithm {
```

Comment:
I prefer to add a class `SamplingBuilder` which handles the enum definitons for all `SamplingAlgorithms` + it's instantiations.

Reply:
add class `SamplingBuilder`